pysal/base.py

```diff
@@ -1,20 +1,120 @@
 """
-Frozen subpackages for meta release.
+Base information for pysal meta package
 """
 
-frozen_packages = {	"libpysal": "4.2.1",
-	"esda": "2.2.0",
-	"giddy": "2.3.0",
-	"inequality": "1.0.0",
-	"pointpats": "2.1.0",
-	"segregation": "1.1.1",
-	"spaghetti": "1.4.0",
-	"mgwr": "2.1.1",
-	"spglm": "1.0.7",
-	"spint": "1.0.6",
-	"spreg": "1.0.4",
-	"spvcm": "0.2.1.post1",
-	"tobler": "0.2.0",
-	"mapclassify": "2.2.0",
-	"splot": "1.1.2"
-}
\ No newline at end of file
+
+federation_hierarchy = {
+    'explore': ['esda', 'giddy', 'segregation',
+                'pointpats', 'inequality',
+                'segregation', 'spaghetti'],
+    'model': ['spreg', 'spglm', 'tobler', 'spint',
+              'mgwr', 'spvcm'],
+    'viz': ['splot', 'mapclassify'],
+    'lib': ['libpysal']
+}
+
+memberships = {'libpysal': 'lib',
+               'esda': 'explore',
+               'giddy': 'explore',
+               'segregation': 'explore',
+               'pointpats': 'explore',
+               'inequality': 'explore',
+               'spaghetti': 'explore',
+               'spreg': 'model',
+               'spglm': 'model',
+               'spint': 'model',
+               'tobler': 'model',
+               'spvcm': 'model',
+               'splot': 'viz',
+               'mapclassify': 'viz'
+}
+
+
+class cached_property(object):
+    """ A property that is only computed once per instance and then replaces
+        itself with an ordinary attribute. Deleting the attribute resets the
+        property.
+
+        Source: https://github.com/bottlepy/bottle/commit/fa7733e075da0d790d809aa3d2f53071897e6f76
+        """
+
+    def __init__(self, func):
+        self.__doc__ = getattr(func, '__doc__')
+        self.func = func
+
+    def __get__(self, obj, cls):
+        if obj is None:
+            return self
+        value = obj.__dict__[self.func.__name__] = self.func(obj)
+        return value
+
+def _installed_version(package):
+    exec(f'import {package}')
+    try:
+        v = eval(f'{package}.__version__')  ## FIX when spvcm is 0.3.0
+    except:
```

Comment:
should we a more explicit here with an `except AttributeError`?

Reply:
or will this be removed following the `spvcm` release?

Reply:
The current installed spvcm doesn't have a version attribute, but once we release 0.3.0 it will.
I think we still want a general check to catch a problem when a package doesn't import, or the version attribute isn't available. In either case setting the table entry to "NA" would flag this